west.yml: Update nRF5 ieee802154 driver to v1.9

jciupis opened the pull request:
This commit introduces nRF IEEE 802.15.4 radio driver version 1.9.

czeslawmakarski committed:
modules: nordic: Introduce nRF IEEE 802.15.4 Service Layer selection

This patch introduces the nRF IEEE 802.15.4 Service Layer configuration.

Signed-off-by: Czeslaw Makarski <Czeslaw.Makarski@nordicsemi.no>

Commit:
west.yml: Update nRF5 ieee802154 driver to v1.9

This commit updates the nRF IEEE 802.15.4 radio driver to version 1.9.

Signed-off-by: Jedrzej Ciupis <jedrzej.ciupis@nordicsemi.no>

hubertmis reviewed (approved):
:+1:

anangl commented on drivers/ieee802154/ieee802154_nrf5.c:
```diff
@@ -173,9 +173,14 @@ static void nrf5_rx_thread(void *arg1, void *arg2, void *arg3)
-	       IEEE802154_HW_TX_RX_ACK | IEEE802154_HW_ENERGY_SCAN |
+	return IEEE802154_HW_FCS |
+	       IEEE802154_HW_FILTER |
+#if !CONFIG_NRF_802154_SL_OPENSOURCE
```
```suggestion
#if !defined(CONFIG_NRF_802154_SL_OPENSOURCE)
```

carlescufi commented:
@hubertmis @anangl this fails on BuildKite with:

```

CMake Error at /workdir/zephyr/cmake/extensions.cmake:401 (add_library):
--
  | No SOURCES given to target:
  | ..__..__var__lib__buildkite-agent__zephyr-module-cache__modules__hal__nordic

```

anangl commented on soc/arm/nordic_nrf/nrf53/Kconfig.defconfig.nrf5340_CPUNET_QKAA:
```diff
@@ -11,4 +11,11 @@ config SOC
 config NUM_IRQS
 	default 30
 
+config IEEE802154_NRF5
+	default y
+	depends on IEEE802154
```
Why not
```
config IEEE802154_NRF5
	default IEEE802154
```
as in `Kconfig.defconfig.nrf5340_CPUAPP_QKAA`?

Commit:
drivers: ieee802154: Add support for IEEE 802.15.4 for nRF5340

This commit introduces support for IEEE 802.15.4 on nRF5340.

Signed-off-by: Jedrzej Ciupis <jedrzej.ciupis@nordicsemi.no>

ioannisg commented on drivers/ieee802154/ieee802154_nrf5.c:
```diff
@@ -53,6 +57,12 @@ static struct nrf5_802154_data nrf5_data;
 #define FRAME_PENDING_BIT (1 << 4)
 
+#if defined(CONFIG_SOC_NRF5340_CPUAPP) || defined(CONFIG_SOC_NRF5340_CPUNET)
+#define EUI64_ADDR (NRF_FICR->INFO.DEVICEID)
```
@anangl how do you ensure that these macros are not accessed by non-secure build?

ioannisg commented on drivers/ieee802154/ieee802154_nrf5.c:
```diff
@@ -53,6 +57,12 @@ static struct nrf5_802154_data nrf5_data;
 #define FRAME_PENDING_BYTE 1
 #define FRAME_PENDING_BIT (1 << 4)
 
+#if defined(CONFIG_SOC_NRF5340_CPUAPP) || defined(CONFIG_SOC_NRF5340_CPUNET)
```
Does the driver software execute on the nRF5340 application core?
Then, does it run on the non-secure domain? If so, I believe the FICR registers are not directly accessible.
Can you clarify?

hubertmis replied:
There are two modules: 
1. 802.15.4 radio driver that is executed in the network core
2. 802.15.4 zephyr shim that is executed in the application core

The radio driver does not use any hardware assigned eui64 addresses. The higher layer is responsible to manage addresses for the driver. However, the shim layer has eui64 in its API. The code you commented is taken from the shim and it is used to create eui64. It is executed in the application core.

I've tested some OpenThread examples and it looks that the shim layer works as expected. I'm not sure if these examples use trustzone or eui64.

Because support for 802.15.4 for nRF53 is for development now, I think we can keep this implementation with a known limitation and fix it right after the upmerge.

Thoughts?

ioannisg replied:
Upstream OpenThread probably run in secure mode. That's why you cannot see the issues.
As it is now, this 802.15.4 zephyr shim layer cannot run in Non-Secure mode. Probably won't compile either.

Can you put some #ifndef CONFIG_TRUSTED_EXECUTION_NONSECURE around the FICR access? (clearly, for nRF5340 CPUAPP only).

hubertmis replied:
@ioannisg , do you have any suggestions how this code could be modified to run in non secure mode? Is there some kind of service to read parts of ficr from non secure environment? I'm asking to plan to fix this in a subsequent pr.

ioannisg replied:
Yes, the Nordic SDK contains secure services to read parts of FICR (or in general, secure memory). There are sample that showcase the usage.
For now, I would suggest to just do not allow the driver to be built for and run in Non-Secure domain, since direct access to FICR is not possible.

czeslawmakarski committed:
drivers: ieee802154: Deny use of nRF FICR registers in NS domain

This commit introduces the catch that prevents building of
IEEE 802.15.4 nRF driver in case the non-secure domain of the core
is used.

Signed-off-by: Czeslaw Makarski <Czeslaw.Makarski@nordicsemi.no>

ioannisg commented:
BTW, @carlescufi the first commit clearly violates the format, so i wonder why compliance does not catch this.

carlescufi commented:
@jciupis  can you change:

`modules/Kconfig.nordic:` - > `modules:`
